File: backend/src/state.ts
import type { Task } from "./models.js";

/** In-memory task store, keyed by task id. */
export class AppState {
  tasks: Map<string, Task> = new Map();

  /** Clears all tasks. Used between test cases. */
  clearTasks(): void {
    this.tasks.clear();
  }
}

File: backend/app/state.py
"""In-memory task store, keyed by task id."""

from __future__ import annotations

from app.models import Task


class AppState:
    def __init__(self) -> None:
        self.tasks: dict[str, Task] = {}

    def clear_tasks(self) -> None:
        """Clears all tasks. Used between test cases."""
        self.tasks.clear()
